operator/webhook/addons/certmanager/main.py

```diff
@@ -0,0 +1,147 @@
+import logging
+import sys
+import os
+from typing import Mapping, List
+
+_LOGGER = logging.getLogger(__name__)
+
+from webhook.core.sync import sync
+
+
+def _new_certificate(parent) -> Mapping:
+    metadata = parent.get("metadata", None)
+    if metadata is None:
+        _LOGGER.error("IntegrationRoute does not contain metadata.")
+        return {}
+
+    annotations = metadata.get("annotations", None)
+    if annotations is None:
+        _LOGGER.error("IntegrationRoute does not contain metadata.annotations.")
+        return {}
+
+    cluster_issuer = annotations.get("cert-manager.io/cluster-issuer", None)
+    if cluster_issuer is None:
+        _LOGGER.error(
+            "IntegrationRoute does not contain metadata.annotations.cert-manager.io/cluster-issuer"
+        )
+        return {}
+
+    common_name = annotations.get("cert-manager.io/common-name", None)
+    if common_name is None:
+        _LOGGER.error(
+            "IntegrationRoute does not contain metadata.annotations.cert-manager.io/common-name"
+        )
+        return {}
+
+    keystore_type = _get_keystore_type(parent)
+    password_secret_ref_name = _get_password_secret_ref_name(parent)
+
+    alt_names = (
+        [n.strip() for n in annotations.get("cert-manager.io/alt-names").split(",")]
+        if annotations.get("cert-manager.io/alt-names", None) is not None
+        else []
+    )
+    organizational_units = (
+        [
+            u.strip()
+            for u in annotations.get(
+                "cert-manager.io/subject-organizationalunits"
+            ).split(",")
+        ]
+        if annotations.get("cert-manager.io/subject-organizationalunits", None)
+        is not None
+        else []
+    )
+    countries = (
+        [
+            c.strip()
+            for c in annotations.get("cert-manager.io/subject-countries").split(",")
+        ]
+        if annotations.get("cert-manager.io/subject-countries", None) is not None
+        else []
+    )
+    provinces = (
+        [
+            p.strip()
+            for p in annotations.get("cert-manager.io/subject-provinces").split(",")
+        ]
+        if annotations.get("cert-manager.io/subject-provinces", None) is not None
+        else []
+    )
+    localities = (
+        [
+            l.strip()
+            for l in annotations.get("cert-manager.io/subject-localities").split(",")
+        ]
+        if annotations.get("cert-manager.io/subject-localities", None) is not None
+        else []
+    )
+
+    dns_names = [
+        f"{annotations['cert-manager.io/common-name']}.{metadata['namespace']}.svc.cluster.local",
+        f"{annotations['cert-manager.io/common-name']}.{metadata['namespace']}.svc",

```

Comment:
Is it worth adding this to be thorough?
```suggestion
        f"{annotations['cert-manager.io/common-name']}.{metadata['namespace']}",

```